openssl-dynamic/src/main/c/ssl.c

```diff
@@ -721,17 +789,29 @@ TCN_IMPLEMENT_CALL(jint, SSL, initialize)(TCN_STDARGS, jstring engine)
                 if (!ENGINE_set_default(ee, ENGINE_METHOD_ALL))
                     err = APR_ENOTIMPL;
             }
-            /* Free our "structural" reference. */
-            if (ee)
-                ENGINE_free(ee);
```

Comment:
looks like this was dropped ... are we now leaking `ee`?

Reply:
Nope what was done before was not really correct. We should only call free in the ssl_init_cleanup

Reply:
So it was moved there

Reply:
can you remove the `ee` intermediate variable then? it introduces the possibility of a leak and makes the ownership less clear.

Reply:
done